### ISSUE TITLE
rclone serve restic ignores --stdio option and serves via tcp/ip instead

### ISSUE THREAD

**Issue by ElRico** [rclone/rclone#6646]
#### The associated forum post URL from `https://forum.rclone.org`

https://forum.rclone.org/t/rclone-v1-61-0-release/34914/3?u=el_rico

#### What is the problem you are having with rclone?

Starting with rclone 1.61.0, the `rclone serve restic` command is ignoring the `--stdio` option and is serving via tcp/ip instead.

#### What is your rclone version (output from `rclone version`)
```
rclone v1.61.0
- os/version: Microsoft Windows 11 Pro 22H2 (64 bit)
- os/kernel: 10.0.22621.963 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.19.4
- go/linking: static
- go/tags: cmount
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

Microsoft Windows 11 Pro 22H2 (64 bit) 

#### Which cloud storage system are you using? (e.g. Google Drive)

Jottacloud

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

.\restic.exe check --read-data-subset=1M --verbose -o rclone.args="serve restic --stdio --b2-hard-delete -vv"

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

```
using temporary cache in <redacted>
rclone: 2022/12/21 15:15:05 DEBUG : rclone: Version "v1.61.0" starting with parameters ["rclone" "serve" "restic" "--stdio" "--b2-hard-delete" "-vv" "<redacted>"]
rclone: 2022/12/21 15:15:05 DEBUG : Creating backend with remote "<redacted>"
rclone: 2022/12/21 15:15:05 DEBUG : Using config file from "<redacted>"
rclone: 2022/12/21 15:15:06 NOTICE: jottacloud root '<redacted>': Serving restic REST API on [http://127.0.0.1:8080/]
rclone: 2022/12/21 15:15:06 DEBUG : file-5577006791947779410: GET request error: object not found
rclone: 2022/12/21 15:15:06 DEBUG : keys: list request
create exclusive lock for repository
rclone: 2022/12/21 15:15:07 DEBUG : locks: list request
rclone: 2022/12/21 15:15:07 DEBUG : locks: list request
rclone: 2022/12/21 15:15:07 DEBUG : snapshots: list request
load indexes
rclone: 2022/12/21 15:15:08 DEBUG : index: list request
check all packs
rclone: 2022/12/21 15:15:09 DEBUG : data: list request
check snapshots, trees and blobs
[0:06] 100.00%  17 / 17 snapshots

read 1048576 bytes of data packs
[0:11] 100.00%  1 / 1 packs

no errors were found
```

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by darthShadow** [comment]
I think this just means that it's using both stdio & the tcp/ip socket. It doesn't seem to be a recent change anyway. Is it causing any issues?

**Comment by ElRico** [comment]
I am using port 8080 to serve another daemon. Until I upgraded to 1.61.0 everything worked fine. Since that upgrade I am getting a fatal error on `rclone serve restic --stdio --b2-hard-delete`, because port 8080 is already in use. Reading the documentation, that seems surprising because `--stdio` should not require port 8080 to be available.
For people who read the docs and have at least passing experience with these things, using `--addr` or the `RCLONE_ADDR` environment variable is an easy fix. For others it might not be that obvious how to fix this.

**Comment by darthShadow** [comment]
Got it. Can you try an older version of rclone too just to be sure if this is a regression or an older bug?

**Comment by ElRico** [comment]
It looks like a regression to me. Here is the same command in 1.60.1:
```
using temporary cache in <redacted>
rclone: 2022/12/21 18:57:20 DEBUG : rclone: Version "v1.60.1" starting with parameters ["rclone" "serve" "restic" "--stdio" "--b2-hard-delete" "-vv" "jottacloud:<redacted>"]
rclone: 2022/12/21 18:57:20 DEBUG : Creating backend with remote "jottacloud:<redacted>"
rclone: 2022/12/21 18:57:20 DEBUG : Using config file from "<redacted>"
rclone: 2022/12/21 18:57:20 DEBUG : jottacloud root '<redacted>': GET /file-5577006791947779410
rclone: 2022/12/21 18:57:20 DEBUG : file-5577006791947779410: GET request error: object not found
rclone: 2022/12/21 18:57:20 DEBUG : jottacloud root '<redacted>': HEAD /config
rclone: 2022/12/21 18:57:21 DEBUG : jottacloud root '<redacted>': GET /keys/
rclone: 2022/12/21 18:57:21 DEBUG : keys: list request
rclone: 2022/12/21 18:57:21 DEBUG : jottacloud root '<redacted>': GET /keys/794cf67ef91142e918744002137e1b209360fd601e7ccbeeb7f3731b852f54b1
rclone: 2022/12/21 18:57:21 DEBUG : jottacloud root '<redacted>': GET /config
create exclusive lock for repository
rclone: 2022/12/21 18:57:21 DEBUG : jottacloud root '<redacted>': GET /locks/
rclone: 2022/12/21 18:57:21 DEBUG : locks: list request
rclone: 2022/12/21 18:57:22 DEBUG : jottacloud root '<redacted>': POST /locks/edde56a08ff0a35e56fa30f325436e34dd770b7764a7eaefc73cfb621edb4924
rclone: 2022/12/21 18:57:22 DEBUG : jottacloud root '<redacted>': GET /locks/
rclone: 2022/12/21 18:57:22 DEBUG : locks: list request
rclone: 2022/12/21 18:57:22 DEBUG : jottacloud root '<redacted>': GET /snapshots/
rclone: 2022/12/21 18:57:22 DEBUG : snapshots: list request
load indexes
rclone: 2022/12/21 18:57:22 DEBUG : jottacloud root '<redacted>': GET /index/
rclone: 2022/12/21 18:57:22 DEBUG : index: list request
rclone: 2022/12/21 18:57:22 DEBUG : jottacloud root '<redacted>': GET /index/51a8a8b62f3239c39e44415b0f32c79d130efc01a3bfd5d10363cba8f8a0c066
check all packs
rclone: 2022/12/21 18:57:24 DEBUG : jottacloud root '<redacted>': GET /data/
rclone: 2022/12/21 18:57:24 DEBUG : data: list request
check snapshots, trees and blobs
rclone: 2022/12/21 18:57:24 DEBUG : jottacloud root '<redacted>': GET /snapshots/b5e6bac2efa8d71e53f800b0f75f647e2442130b631be54b0bd3a48345066e54
rclone: 2022/12/21 18:57:24 DEBUG : jottacloud root '<redacted>': GET /snapshots/795574d7f50eeae5423ee060a2cf95bd4b1445908b092befe8eccdaec5683917
rclone: 2022/12/21 18:57:24 DEBUG : jottacloud root '<redacted>': GET /snapshots/501dbe35a1506427c11d9a8d137b62413853395a7bdac46b4fbbd0de58783cfb
rclone: 2022/12/21 18:57:24 DEBUG : jottacloud root '<redacted>': GET /snapshots/fd708995ab8faa9a7db2393f44358e43343c5dbfdcc497fafd5eca94635ad526
rclone: 2022/12/21 18:57:24 DEBUG : jottacloud root '<redacted>': GET /snapshots/2a0705d2e0b5d4a782e10526febdf9ca1f4013ebf8384e864c257ef11e8366e3
rclone: 2022/12/21 18:57:24 DEBUG : jottacloud root '<redacted>': GET /snapshots/095d4b2ae3c6bb34067022430e452e5c8648ab1823a4b5966b27b0fc374fd086
rclone: 2022/12/21 18:57:25 DEBUG : jottacloud root '<redacted>': GET /snapshots/1f5c182038f2c9894b811cbdeeeef13dff031c67796bf50bfd6c742ccbef94bc
rclone: 2022/12/21 18:57:25 DEBUG : jottacloud root '<redacted>': GET /snapshots/ef86e33bbf2caad6f339281dd025a1f8e30a24b60ef3e06e52902bb3aca6b1a7
rclone: 2022/12/21 18:57:25 DEBUG : jottacloud root '<redacted>': GET /snapshots/841a4b0c25ab9dd8ba2a0d31127beaef73df088fedc3a51e6b98a0e89e57f9cc
rclone: 2022/12/21 18:57:25 DEBUG : jottacloud root '<redacted>': GET /snapshots/86d646a670f817a1bb2f8e5f709cb6a6c735e364401a7f7e6d0df4264442bee0
rclone: 2022/12/21 18:57:25 DEBUG : jottacloud root '<redacted>': GET /snapshots/7e18f956682ae2c8cb274bd5f1f6d61010e611ef7178f33392550851a0ff9366
rclone: 2022/12/21 18:57:25 DEBUG : jottacloud root '<redacted>': GET /snapshots/55345b09c48250f7ff3618c2468b109b186ec894a697dae3e897c42334701ba6
rclone: 2022/12/21 18:57:25 DEBUG : jottacloud root '<redacted>': GET /snapshots/0c07e90ec501ac6e90f88055ff0f9aaf4a20cf09f9ca95267f1efe0d12da5508
rclone: 2022/12/21 18:57:25 DEBUG : jottacloud root '<redacted>': GET /snapshots/60d2521ffd42f425a6af925ae6f046fe5babea59c82edc17ac0b0b36c42d2517
rclone: 2022/12/21 18:57:25 DEBUG : jottacloud root '<redacted>': GET /snapshots/781589ba2c8887641c8741d1302eaae540bdff4b1972d98a25578ead313dbb43
rclone: 2022/12/21 18:57:25 DEBUG : jottacloud root '<redacted>': GET /snapshots/8ac0105f304b7993970dbc2a64954fb4e095cbce838977117f51fe367c343029
rclone: 2022/12/21 18:57:25 DEBUG : jottacloud root '<redacted>': GET /snapshots/29bfc3f7a107e9a88bdc307adbe9f86d8780d5275fd3bb8cb42d87bbb008cf3b
rclone: 2022/12/21 18:57:25 DEBUG : jottacloud root '<redacted>': GET /data/e2c04959636e0fbd89718a76e96a5f36a49a46aba5fcd3435980ef360d554547
rclone: 2022/12/21 18:57:25 DEBUG : jottacloud root '<redacted>': GET /data/81bfe5900f307b8df2d474425daf2b4e1a061df6a1694f39a7f7c27640d22cc3
rclone: 2022/12/21 18:57:31 DEBUG : jottacloud root '<redacted>': GET /data/3d0cd55908ca16d74fe26ded0676c1f2396129c4602097e5c35bede987c362e4
[0:06] 100.00%  17 / 17 snapshots

read 1048576 bytes of data packs
[0:00] 100.00%  1 / 1 packs

no errors were found
rclone: 2022/12/21 18:57:31 DEBUG : jottacloud root '<redacted>': DELETE /locks/edde56a08ff0a35e56fa30f325436e34dd770b7764a7eaefc73cfb621edb4924
```

It looks a bit different, but does not mention serving via `http://127.0.0.1:8080/` anywhere.

**Comment by darthShadow** [comment]
Ah, yes, I see the regression in the code now.

Once [fix-6644-serve-restic](https://github.com/rclone/rclone/tree/fix-6644-serve-restic?rgh-link-date=2022-12-21T12%3A35%3A40Z) is merged, it should be fixed.

**Comment by ncw** [comment]
I think this should fix the problem @ElRico - can you give it a test please

[v1.62.0-beta.6660.8863fb491.fix-6644-serve-restic](https://beta.rclone.org/branch/fix-6644-serve-restic/v1.62.0-beta.6660.8863fb491.fix-6644-serve-restic/) on branch [fix-6644-serve-restic](https://github.com/rclone/rclone/tree/fix-6644-serve-restic) (uploaded in 15-30 mins)

**Comment by ncw** [comment]
This fix has been released in v1.61.1 which is available now

**Comment by ElRico** [comment]
This is just to confirm that 1.61.1 does indeed fix the issue on my system. Thank you for the fix @ncw!
